src/Stream.php

```
@@ -320,23 +318,14 @@ public function getMetadata($key = null)
      */
     private function setStream($stream, string $mode = 'r'): void
     {
-        $error    = null;
         $resource = $stream;
 
         if (is_string($stream)) {
-            set_error_handler(function ($e) use (&$error) {
-                if ($e !== E_WARNING) {
-                    return;
-                }
-
-                $error = $e;
-            });
-            $resource = fopen($stream, $mode);
-            restore_error_handler();
-        }
-
-        if ($error) {
-            throw new Exception\RuntimeException('Invalid stream reference provided');
+            try {
+                $resource = fopen($stream, $mode);
+            } catch (Throwable $e) {
+                throw new Exception\RuntimeException('Invalid stream reference provided');
```

We should really rethrow the previous exception

---

s/rethrow/include

---

Do you think it's a reasonable change - tests are now passing locally for me on 8.x
… The previous behaviour was to throw what's there - I'll amend to add previous perhaps? The integration tests would fail with anything other than a RuntimeException afaict

---

I think this is really only in case of a stream error - an exception should be rare, but is a healthy approach to a failure.